src/util/api/users/getPublicInfo.js

```diff
@@ -1,19 +1,37 @@
 import {
+  API_EXTERNAL_HOSTNAME,
   AVATAR_DEFAULT_PATH,
   CIVIC_LIKER_START_DATE,
   SUBSCRIPTION_GRACE_PERIOD,
 } from '../../../constant';
+import { ValidationError } from '../../ValidationError';
+import {
+  checkAddressValid,
+  checkCosmosAddressValid,
+} from '../../ValidationHelper';
 import {
   userCollection as dbRef,
 } from '../../firebase';
 
-export function formatUserCivicLikerProperies(id, data) {
-  const { avatar, civicLiker } = data;
+
+function isValidUserDoc(userDoc) {
+  if (!userDoc || !userDoc.exists) {
+    return false;
+  }
+  const userData = userDoc.data();
+  if (userData.isDeleted) {
+    return false;
+  }
+  return true;
+}
+
+function formatUserCivicLikerProperies(userDoc) {
+  const { id } = userDoc;
+  const data = userDoc.data();
+  const { civicLiker } = data;
   const payload = data;
   payload.user = id;
-  if (!avatar) {
-    payload.avatar = AVATAR_DEFAULT_PATH;
-  }
+  payload.avatar = `https://${API_EXTERNAL_HOSTNAME}/users/id/${id}/avatar`;
```

Comment:
Can we add `size` qs with default value here to hint there is a size configurable?
define `const DEFAULT_AVATAR_SIZE = 400` in `constant and
```suggestion
  payload.avatar = `https://${API_EXTERNAL_HOSTNAME}/users/id/${id}/avatar?size=${DEFAULT_AVATAR_SIZE}`;
```